.github/workflows/job_build_windows.yml

```diff
@@ -82,6 +89,14 @@ jobs:
           self-hosted-runner: 'true'
           show-cache-info: 'true'
 
+      - name: Prepare Intel OneAPI Toolkit
+        if: ${{ inputs.compiler-type }} == "DPCPP"
+        id: oneapi_install
+        run: |
+            Invoke-WebRequest -Uri '${{ env.ONEAPI_TOOLKIT_LINK }}' -OutFile 'oneapi_installer.exe'
+            .\oneapi_installer.exe -s -a --eula accept --install-dir ${{ env.ONEAPI_INSTALL_DIR }} --components intel.oneapi.win.cpp-dpcpp-common
+            ${{ env.ONEAPI_INSTALL_DIR }}\setvars.bat
```

Comment:
if you call setupvars in this step, it will not be shared with steps below.
You need to call it in cmake configre  stage